frontend/dockerfile/dockerfile_lint_test.go

```diff
@@ -302,12 +365,18 @@ func checkUnmarshal(t *testing.T, sb integration.Sandbox, c *client.Client, dir
 			},
 			Frontend: "dockerfile.v0",
 		})
-		require.NoError(t, err)
+		if err != nil {
+			return nil, err
+		}
 
 		lintResults, err := unmarshalLintResults(res)
 		require.NoError(t, err)
 
-		require.Equal(t, len(expected), len(lintResults.Warnings))
+		if lintResults.Error != nil {
+			require.Equal(t, lintTest.UnmarshalBuildErr, lintResults.Error.Message)
```

Comment:
Also check that the `lintResults.Error.Location.SourceIndex >= 0` and that the index exist and I'm not sure if I see checks for `lintResults.Sources` being present atm.

Reply:
Added!

Reply:
Is the `lintResults.Sources` check somewhere else? Merging this as not directly related, but follow-up if it is unchecked atm.